File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/DefaultKinveyClientCallback.java
package com.kinvey.androidTest.callback;

import com.kinvey.androidTest.model.Person;
import com.kinvey.java.core.KinveyClientCallback;

import java.util.concurrent.CountDownLatch;

/**
 * Created by yuliya on 09/14/17.
 */

public class DefaultKinveyClientCallback implements KinveyClientCallback<Person> {

    private CountDownLatch latch;
    private Person result;
    private Throwable error;

    public DefaultKinveyClientCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    @Override
    public void onSuccess(Person result) {
        this.result = result;
        finish();
    }

    @Override
    public void onFailure(Throwable error) {
        this.error = error;
        finish();
    }

    void finish() {
        latch.countDown();
    }

    public Person getResult() {
        return result;
    }

    public Throwable getError() {
        return error;
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/DefaultKinveyClientCallback.kt
package com.kinvey.androidTest.callback

import com.kinvey.androidTest.model.Person
import com.kinvey.java.core.KinveyClientCallback
import java.util.concurrent.CountDownLatch

/**
 * Created by yuliya on 09/14/17.
 */

class DefaultKinveyClientCallback(private val latch: CountDownLatch) : KinveyClientCallback<Person> {
    var result: Person? = null
        private set
    var error: Throwable? = null
        private set

    override fun onSuccess(result: Person) {
        this.result = result
        finish()
    }

    override fun onFailure(error: Throwable) {
        this.error = error
        finish()
    }

    internal fun finish() {
        latch.countDown()
    }

}